src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -470,20 +475,52 @@ private boolean doSaveImpl(final String path,
                @Override
                public void onError(ServerError error)
                {
+                  // Always log save errors.
                   Debug.logError(error);
+                  
+                  // Inform the user once if this was an autosave failure.
+                  if (isAutosave && !loggedAutosaveError)
+                  {
+                     loggedAutosaveError = true;
+
+                     RStudioGinjector.INSTANCE.getGlobalDisplay().showErrorMessage(
+                           "Error Autosaving File",
+                           "RStudio was unable to autosave this file. You may need " +
+                           "to restart RStudio.");
+                  }
+                 
+                  // Avoid reporting auto-save errors to the user via the indicator,
+                  // as this could lead to a relentless cascade of errors while the
+                  // user attempts to edit the document.
                   if (progress != null)
-                     progress.onError(error.getUserMessage());
+                  {
+                     if (isAutosave)
+                     {
+                        progress_.onCompleted();
```

Comment:
Not sure about treating this as a completion. One consequence of this is that the rest of the code is going to think that changes are backed up when they aren't. Will this cause the user to not get the "you haven't saved your changes" warning?

Reply:
This is only for the progress indicator, not for any logic related to successful document save (ie: we don't touch the `onCompleted_` callback). This change ensures that the progress indicator is dismissed followed a failed save.

Reply:
Sorry, I'm wrong: different kinds of progress indicators get used in different calls here. Still, it seems like we need a way to indicate an indicator error without always notifying the user.